### PR TITLE
Remove agl folder

### PR DESCRIPTION
**Pull request by ThePixelGamer**
This PR removes the agl folder, fixes include inconsistencies and adds a proper ordering for the includes.
Currently that looks like this:
 - regular <> includes
 - nn/nvn/vapours (folders from NintendoSDK) 
 - files that start with sead 
 - regular "" includes

- - -
This change is [Reviewable](https://reviewable.io/reviews/open-ead/agl/12)

### CODE REVIEW

File: include/common/aglGPUMemAddr.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
-#include "agl/detail/aglMemoryPoolHeap.h"
 #include "aglGPUMemBlock.h"
```
Comment (MonsterDruide1) [inline include/common/aglGPUMemAddr.h]: Should this also get a fully qualified name, or stay like this? What about the ordering in such a case?

Comment (ThePixelGamer) [reply include/common/aglGPUMemAddr.h]: I personally prefer the styling of not using folders when pulling files from the same folder but if leo wants I can change to include the ``common/`` for consistency

**Review by MonsterDruide1 — COMMENTED** [review]
> Reviewed 64 of 64 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/open-ead/agl/12)* status: all files reviewed, 4 unresolved discussions (waiting on @leoetlino and @ThePixelGamer)

___
*[`.clang-format` line 40 at r1](https://reviewable.io/reviews/open-ead/agl/12#-NP9EOBbB48YFj-XVllt:-NP9EOBbB48YFj-XVllu:b-uyp46t) ([raw file](https://github.com/open-ead/agl/blob/177a9370137ff4304f82a3f28ce00e5415a1cb98/.clang-format#L40)):*
> ```YAML
>     Priority:        3
>   - Regex:           '^<'
>     Priority:        1
> ```

Order it according to priority in here as well?



_Code quote:_
```YAML
  - Regex:           '^<'
    Priority:        1
```
___
*[`include/common/aglRenderTarget.h` line 3 at r1](https://reviewable.io/reviews/open-ead/agl/12#-NP9FmVBEdgtsWAQaDG9:-NP9FmVBEdgtsWAQaDGA:b4btxug) ([raw file](https://github.com/open-ead/agl/blob/177a9370137ff4304f82a3f28ce00e5415a1cb98/include/common/aglRenderTarget.h#L3)):*
> ```objc
> #pragma once
> 
> #include "aglTextureData.h"
> ```

Same here - including a path, maybe?
___
*[`include/texture/aglTextureDataSerializer.h` line 13 at r1](https://reviewable.io/reviews/open-ead/agl/12#-NP9G0OXDuDH-S7LcsU4:-NP9G0OXDuDH-S7LcsU5:b-ju0c8a) ([raw file](https://github.com/open-ead/agl/blob/177a9370137ff4304f82a3f28ce00e5415a1cb98/include/texture/aglTextureDataSerializer.h#L13)):*
> ```objc
> };
> 
> }  // namespace agl
> ```

Oh, look - a missing newline at the end of the file! If we're touching nearly every file here, fix that?

**Comment by ThePixelGamer** [comment]
> Order it according to priority in here as well?
> 
> ```yaml
>   - Regex:           '^<'
>     Priority:        1
> ```

nada in this case the sead files will be grouped with this regex if it's at the top (at least from my testing but I might've done something wrong?)

**Comment by ThePixelGamer** [comment]
*[`include/texture/aglTextureDataSerializer.h` line 13 at r1](https://reviewable.io/reviews/open-ead/agl/12#-NP9G0OXDuDH-S7LcsU4:-NP9Hokm2pM3P3jP-6df:b-896fix) ([raw file](https://github.com/open-ead/agl/blob/177a9370137ff4304f82a3f28ce00e5415a1cb98/include/texture/aglTextureDataSerializer.h#L13)):*
<details><summary><i>Previously, MonsterDruide1 wrote…</i></summary><blockquote>

Oh, look - a missing newline at the end of the file! If we're touching nearly every file here, fix that?
</blockquote></details>

Done.

**Review by ThePixelGamer — COMMENTED** [review]
> Dismissed @MonsterDruide1 from a discussion.
> *[Reviewable](https://reviewable.io/reviews/open-ead/agl/12)* status: 59 of 64 files reviewed, 3 unresolved discussions (waiting on @leoetlino and @MonsterDruide1)

**Review by MonsterDruide1 — COMMENTED** [review]
> Reviewed 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/open-ead/agl/12)* status: all files reviewed, 2 unresolved discussions (waiting on @leoetlino and @ThePixelGamer)

___
*[`.clang-format` line 40 at r1](https://reviewable.io/reviews/open-ead/agl/12#-NP9EOBbB48YFj-XVllt:-NP9Hz3-8UTYJjuF5dqe:b1vt732) ([raw file](https://github.com/open-ead/agl/blob/177a9370137ff4304f82a3f28ce00e5415a1cb98/.clang-format#L40)):*
<details><summary><i>Previously, MonsterDruide1 wrote…</i></summary><blockquote>

Order it according to priority in here as well?
</blockquote></details>

Oh, good point. Nevermind then.

**Comment by aboood40091** [comment]
Please use <> for includes not in the same folder. Also, the folders need renaming. Are you going to do that in a separate PR?

**Comment by ThePixelGamer** [comment]
Which folders need renaming? If you could send it in the discord

**Comment by ThePixelGamer** [comment]
Fixed and waiting for leo's input for the <> suggestion

**Review by MonsterDruide1 — COMMENTED** [review]
> Reviewed 32 of 32 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/open-ead/agl/12)* status: all files reviewed, 3 unresolved discussions (waiting on @leoetlino and @ThePixelGamer)

___
*[`src/utility/aglParameterStringMgr.cpp` line 9 at r3](https://reviewable.io/reviews/open-ead/agl/12#-NP9dIOD70rfjJvylx6w:-NP9dIOE4Rifo0R_IyeY:bpq24yv) ([raw file](https://github.com/open-ead/agl/blob/60b58b904341618f9576b1782474ad60d8a535ef/src/utility/aglParameterStringMgr.cpp#L9)):*
> ```cpp
> ParameterStringMgr::ParameterStringMgr() {
> #ifdef SEAD_DEBUG
>     setNodeName("utility::ParameterStringMgr");
> ```

Was this an intended change?

**Comment by ThePixelGamer** [comment]
Oops, was abusing vscode's replace feature

**Comment by ThePixelGamer** [comment]
*[`src/utility/aglParameterStringMgr.cpp` line 9 at r3](https://reviewable.io/reviews/open-ead/agl/12#-NP9dIOD70rfjJvylx6w:-NP9etwC3uBEfFWZOV2T:b-896fix) ([raw file](https://github.com/open-ead/agl/blob/60b58b904341618f9576b1782474ad60d8a535ef/src/utility/aglParameterStringMgr.cpp#L9)):*
<details><summary><i>Previously, MonsterDruide1 wrote…</i></summary><blockquote>

Was this an intended change?
</blockquote></details>

Done.

**Review by MonsterDruide1 — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/open-ead/agl/12)* status: all files reviewed, 2 unresolved discussions (waiting on @leoetlino, @MonsterDruide1, and @ThePixelGamer)

**Review by MonsterDruide1 — COMMENTED** [review]
> Reviewed 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/open-ead/agl/12)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @leoetlino)

**Pull request merged**